### PR TITLE
WindowProxy: add same-origin indexed property tests

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/html/issues/2541.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5596)

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5596/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223104794)

# Lint #

Passed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223104798)

# Firefox (nightly channel) #
Testing web-platform-tests at revision c0baff827815b25a20793bae223c854f46098e8f
Using browser at version BuildID 20170417100236; SourceStamp c697e756f738ce37abc56f31bfbc48f55625d617
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5596/html/browsers/the-window-object/window-indexed-properties-strict.html">/html/browsers/the-window-object/window-indexed-properties-strict.html</a></summary>


|                          Subtest                          | Results | Messages |
|-----------------------------------------------------------|---------|----------|
|                                                           | OK      |          |
| `Indexed properties of the window object (strict mode)`   | PASS    |          |
| `Indexed properties of the window object (strict mode) 1` | PASS    |          |
| `Indexed properties of the window object (strict mode) 2` | PASS    |          |
| `Indexed properties of the window object (strict mode) 3` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5596/html/browsers/the-window-object/window-indexed-properties.html">/html/browsers/the-window-object/window-indexed-properties.html</a></summary>


|                            Subtest                            | Results | Messages |
|---------------------------------------------------------------|---------|----------|
|                                                               | OK      |          |
| `Indexed properties of the window object (non-strict mode)`   | PASS    |          |
| `Indexed properties of the window object (non-strict mode) 1` | PASS    |          |
| `Indexed properties of the window object (non-strict mode) 2` | PASS    |          |
| `Indexed properties of the window object (non-strict mode) 3` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223104799)

# Chrome (unstable channel) #
Testing web-platform-tests at revision c0baff827815b25a20793bae223c854f46098e8f
Using browser at version 59.0.3067.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5596/html/browsers/the-window-object/window-indexed-properties-strict.html">/html/browsers/the-window-object/window-indexed-properties-strict.html</a></summary>


|                          Subtest                          | Results |                                       Messages                                       |
|-----------------------------------------------------------|---------|--------------------------------------------------------------------------------------|
|                                                           | OK      |                                                                                      |
| `Indexed properties of the window object (strict mode)`   | PASS    |                                                                                      |
| `Indexed properties of the window object (strict mode) 1` | FAIL    | `assert_throws: function "function () {\n    window[0] = "foo";\n  }" did not throw` |
| `Indexed properties of the window object (strict mode) 2` | FAIL    | `assert_throws: function "function () {\n    window[1] = "foo";\n  }" did not throw` |
| `Indexed properties of the window object (strict mode) 3` | PASS    |                                                                                      |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5596/html/browsers/the-window-object/window-indexed-properties.html">/html/browsers/the-window-object/window-indexed-properties.html</a></summary>


|                            Subtest                            | Results |                                              Messages                                              |
|---------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------------|
|                                                               | OK      |                                                                                                    |
| `Indexed properties of the window object (non-strict mode)`   | PASS    |                                                                                                    |
| `Indexed properties of the window object (non-strict mode) 1` | FAIL    | `assert_throws: function "() => Object.defineProperty(window, 0, { value: "bar" })" did not throw` |
| `Indexed properties of the window object (non-strict mode) 2` | FAIL    | `assert_throws: function "() => Object.defineProperty(window, 0, { value: "bar" })" did not throw` |
| `Indexed properties of the window object (non-strict mode) 3` | PASS    |                                                                                                    |

</details>
</details>

File: html/browsers/the-window-object/window-indexed-properties-strict.html [hunk 1]
```diff
@@ -23,6 +23,7 @@
   assert_throws(new TypeError(), function() {
     window[0] = "foo";
   });
+  assert_throws(new TypeError(), () => Object.defineProperty(window, 0, { value: "bar" }))
```
Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties-strict.html]: Shouldn't we also be testing properties which don't have anything in them, as that was part of the original bug report?

Comment (annevk) [inline html/browsers/the-window-object/window-indexed-properties-strict.html]: The added line below was supposed to write to property 1, but I copypasta'd.

Comment (annevk) [inline html/browsers/the-window-object/window-indexed-properties-strict.html]: Should be fixed now.

**Comment by yuki3** [comment]
I don't know well about WPT, so I guess window.length == 0 is obvious in these tests, right?  Otherwise, I'd like to check it.  LGTM.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/web-platform-tests/5596#-:-Ki0MlAc_pka62Jf8DkF:bkxxzmp)*

**Comment by annevk** [comment]
window.length is 1 because there's an `<iframe>` in the HTML. That's why self[0] is assumed to return something whereas self[1] is used to return undefined.

**Review by domenic — APPROVED** [review]

**Comment by yuki3** [comment]
Ah, I see.  Thanks for the explanation.  LGTM.

**Pull request merged**